### DISCUSSION TITLE
Setting `unicorn/consistent-function-scoping` on util file

### DISCUSSION THREAD

**Discussion by eneeio-nm** [antfu/eslint-config#605] [Q&A] [answered]
Hi all,

I'm in the process of moving to this linting option (currently using v8 of eslint and Prettier).

Let's say I have a simple util file that contains the following:

```ts
export function exampleFactory() {
  function exampleFunction() {
    return {
      example: true
    };
  }

  return {
    exampleFunction
  };
}
```

When I run `npm run lint` the following is reported for the `exampleFunction` function:

> Move function 'exampleFunction' to the outer scope. `eslint(unicorn/consistent-function-scoping)`

I experience similar errors with my Vue composables, but I have got round those by creating a 'rule' for the folder:

```
{
  files: ['**/composables/*.ts'],
  rules: {
    'unicorn/consistent-function-scoping': 'off'
  }
}
```

Granted, I could add my `utils` folder to the above, but before doing that I wanted to reach out to find out if there's a better way of handling this situation.

**Comment by Dimava** [comment]
Rule of thumb: if a rule doesn't make sense and you have 3 eslint-ignores for it, just disable it

> A function definition should be placed as close to the top-level scope as possible without breaking its captured values. **This improves readability**

If the rule doesn't work for you because you want to do *exactly* what it was made to prevent (in your case factory functions with inline function definitions) you don't need it

---
## Fail

```js
export function doFoo(foo) {
	// Does not capture anything from the scope, can be moved to the outer scope
	function doBar(bar) {
		return bar === 'bar';
	}

	return doBar;
}

function doFoo(foo) {
	const doBar = bar => {
		return bar === 'bar';
	};
}
```

## Pass

```js
function doBar(bar) {
	return bar === 'bar';
}

export function doFoo(foo) {
	return doBar;
}

export function doFoo(foo) {
	function doBar(bar) {
		return bar === 'bar' && foo.doBar(bar);
	}

	return doBar;
}
```

**Comment by Dimava** [comment, marked as answer]
@eneeio-nm this rule was killed in https://github.com/antfu/eslint-config/releases/tag/v3.5.0

**Reply by eneeio-nm** [reply]
Ooo, good to know. I'll revisit this when I get chance. Thanks for the heads-up 👍
